test/index.test.ts

```diff
@@ -20,121 +21,127 @@ describe('gcpMetadata', () => {
+    try {
+      await gcp.instance();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert.strictEqual(e.message, 'Unsuccessful response status code');
+    }
   });
 
-  it('should return error when res is empty', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should return error when res is empty', async () => {
     nock(HOST).get(`${PATH}/${TYPE}`).reply(200, null);
-    getMetadata((err) => {
-      assert(err instanceof Error);
-      done();
-    });
+    try {
+      await gcp.project();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
   });
 
-  it('should return error when flavor header is incorrect', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should return error when flavor header is incorrect', async () => {
     nock(HOST).get(PATH).reply(200, {}, {'Metadata-Flavor': 'Hazelnut'});
-    getMetadata((err) => {
-      assert(err instanceof Error);
-      done();
-    });
+    try {
+      await gcp.project();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
   });
 
-  it('should return error if statusCode is not 200', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should return error if statusCode is not 200', async () => {
     nock(HOST).get(PATH).reply(418, {}, {'Metadata-Flavor': 'Google'});
-    getMetadata((err) => {
-      assert(err instanceof Error);
-      done();
-    });
+    try {
+      await gcp.project();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
   });
 
-  it('should retry if the initial request fails', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should retry if the initial request fails', async () => {
     nock(HOST).get(`${PATH}/${TYPE}`).reply(500);
     nock(HOST).get(`${PATH}/${TYPE}`).reply(200, {});
-    getMetadata((err, res) => {
-      assert.equal(res!.config.url, `${BASE_URL}/${TYPE}`);
-      assert.equal(res!.config.headers['Metadata-Flavor'], 'Google');
-      done();
-    });
+    const res = await gcp.instance();
+    assert.equal(res.config.url, `${BASE_URL}/${TYPE}`);
+    assert.equal(res.config.headers['Metadata-Flavor'], 'Google');
   });
 
-  it('should throw if request options are passed', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
-    // tslint:disable-next-line no-any
-    (gcpMetadata as any).instance({qs: {one: 'two'}}, (err: Error) => {
-      assert.notEqual(err, null);
-      assert(err.message.startsWith('\'qs\' is not a valid'));
-      done();
-    });
+  it('should throw if request options are passed', async () => {
+    try {
+      // tslint:disable-next-line no-any testing validation
+      await (gcp as any).instance({qs: {one: 'two'}});
+      assert.fail('Expected to throw');
```

Comment:
nit: This assert will be caught and the error message reported to the developer would not immediately point to the correct assertion that failed.